cmd/antrea-agent/util.go

```diff
@@ -79,3 +84,25 @@ func parsePortRange(portRangeStr string) (start, end int, err error) {
 
 	return start, end, nil
 }
+
+func getServiceListOptionsModifier(serviceProxyName string) (func(*metav1.ListOptions), error) {
+	var serviceNameSelector, nonHeadlessServiceSelector *labels.Requirement
+	var err error
+	nonHeadlessServiceSelector, err = labels.NewRequirement(corev1.IsHeadlessService, selection.DoesNotExist, nil)
```

Comment:
Will this work as expected? I don't think a headless Service will have this label automatically. Could you double check?
```
	// IsHeadlessService is added by Controller to an Endpoint denoting if its parent
	// Service is Headless. The existence of this label can be used further by other
	// controllers and kube-proxy to check if the Endpoint objects should be replicated when
	// using Headless Services
	IsHeadlessService = "service.kubernetes.io/headless"
```

Reply:
Yes, I understand Endpoints will have this label for headless Service, but here you are trying to filter Services. I am asking will this label be added automatically for a headless Service?

Reply:
This is a filter for informer, which means that it will filter out either Endpoint or Service.

Reply:
>  I am asking will this label be added automatically for a headless Service?

No